forest/config.py

```diff
@@ -21,13 +21,30 @@
 
 """
 import os
+import string
 import yaml
 from forest.export import export
 
 
 __all__ = []
 
 
+def combine_variables(os_environ, args_variables):
+    """Utility function to update environment with user-specified variables
+
+    .. note: When there is a key clash the user-specified args take precedence
+
+    :param os_environ: os.environ dict
+    :param args_variables: variables parsed from command line
+    :returns: merged dict
+    """
+    if args_variables is None:
+        return dict(os_environ)
+    variables = dict(os_environ)
+    variables.update(dict(args_variables))
+    return variables
```

Comment:
We can avoid the repetition and the unnecessary `is None`:
```python
variables = dict(os.environ)
if args_variables:
    variables.update(...)
return variables
```

Reply:
Agreed, that is a nice refactor

Reply:
Ah, just spotted you're using the old falsey empty list/None check. I've gone with explicit `is not None` since `[]` works with `dict.update`. I'm trying my best to split a few more hairs :wink:

Reply:
Why would you want to update the dict with an empty list?!